core/core.go

```diff
@@ -65,7 +69,7 @@ type Driver interface {
 type NetworkDriver interface {
 	// A network driver implements the programming logic for network
 	Driver
-	Init(config *Config, stateDriver StateDriver) error
+	Init(config *Config, info *InstanceInfo, stateDriver StateDriver) error
```

Comment:
@mapuri - not sure if we should combine the Config and Instance info as init metadata structure; To avoid changing the core interfaces, if some other data needs to be passed during init, we may end up adding one more field to it.